modules/evm/src/runner/handler.rs

```diff
@@ -304,7 +305,14 @@ impl<'vicinity, 'config, 'meter, T: Config> HandlerT for Handler<'vicinity, 'con
 	}
 
 	fn code(&self, address: H160) -> Vec<u8> {
-		Pallet::<T>::code_at_address(&address)
+		let mut token = [0u8; 19];
+		token[16] = 1;
+		// 0x0000000000000000000000000000000001000000
+		if address.as_bytes().starts_with(&token) {
+			Pallet::<T>::code_at_address(&H160::from_str("0x0000000000000000000000000000000000000800").unwrap())
```

Comment:
should make this a `const` if possible, and have some comments explain why
also needs to update code_size & code_hash